src/deferred.js

```diff
@@ -0,0 +1,63 @@
+
+var oldDeferred = jQuery.Deferred,
+	tuples = [
+		// action, add listener, callbacks, .then handlers, final state
+		[ "resolve", "done", jQuery.Callbacks("once memory"),
+			jQuery.Callbacks("once memory"), "resolved" ],
+		[ "reject", "fail", jQuery.Callbacks("once memory"),
+			jQuery.Callbacks("once memory"), "rejected" ],
+		[ "notify", "progress", jQuery.Callbacks("memory"),
+			jQuery.Callbacks("memory") ]
+	];
+
+jQuery.Deferred = function( func ) {
+    var deferred = oldDeferred(),
+		promise = deferred.promise();
+
+	deferred.pipe = promise.pipe =  function( /* fnDone, fnFail, fnProgress */ ) {
```

Comment:
There's an extra space after the second "=", and the indentation looks wrong from here down.